app/controllers/homepage_controller.rb

```diff
@@ -51,21 +51,23 @@ def index
         raise ArgumentError.new("Unknown view_type #{@view_type}")
       end
 
-    search_result = find_listings(params, per_page, compact_filter_params, includes.to_set)
+    main_search = (feature_enabled?(:location_search) && search_engine == :zappy) ? MarketplaceService::API::Api.configurations.get(community_id: @current_community.id).data[:main_search] : :keyword
+    location_search_in_use = (feature_enabled?(:location_search) && search_engine == :zappy && main_search == :location)
```

Comment:
These two lines seem to have a lot of repetition and the logic feels a bit too complicated. Can we figure out a way to make it simpler?